ilastik/applets/dataSelection/opDataSelection.py

```diff
@@ -924,9 +945,54 @@ def addLane(self, laneIndex):
         # We might be called from within the context of our own insertSlot signal.
         if numLanes == laneIndex:
             super(OpMultiLaneDataSelectionGroup, self).addLane(laneIndex)
+        return self.getLane(laneIndex)
 
     def removeLane(self, laneIndex, finalLength):
         """Reimplemented from base class."""
         numLanes = len(self.innerOperators)
         if numLanes > finalLength:
             super(OpMultiLaneDataSelectionGroup, self).removeLane(laneIndex, finalLength)
+
+    @property
+    def workflow(self) -> Workflow:
+        return self.parent
+
+    @property
+    def role_names(self) -> List[str]:
+        return self.DatasetRoles.value
+
+    def pushLane(self, role_infos: Dict[str, DatasetInfo]):
+        original_num_lanes = self.num_lanes
+        try:
+            lane = self.addLane(self.num_lanes)
+            lane.configure(infos=role_infos)
+            self.workflow.handleNewLanesAdded()
+        except Exception as e:
+            self.removeLane(original_num_lanes, original_num_lanes)
+            raise e
+
+    def popLane(self):
+        self.removeLane(self.num_lanes -1, self.num_lanes - 1)
+
+    @property
+    def num_lanes(self) -> int:
+        return len(self.innerOperators)
+
+    def get_lane(self, lane_idx: int) -> OpDataSelectionGroup:
+        return self.innerOperators[lane_idx]
+
+    def get_role_info(self, lane_idx: int, role: str) -> Optional[DatasetInfo]:
+        info_slot = self.DatasetGroup[lane_idx][self.role_names.index(role)]
+        return info_slot.value if info_slot.ready() else None
+
+    def get_last_axes_keys(self) -> Dict[str, Optional[AxisTags]]:
```

Comment:
"...and naming things". Here I would rather have a more descriptive name. So either something that does not imply the implementation ("last") or that more clearly states that this method gets the axistags of the last lane. So maybe either `get_axes_keys_last_lane`. Or make this method more generic, like `get_axis_keys(self, lane_idx)` and allow a `-1` for lane-index. That way the gui implementation of a guess doesn't dripple down into the operator. Does that make sense?

Reply:
Makes total sense. In fact, I think I eventually reached the same conclusion as I was fixing the code some more.